plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1044 @@
+            result['after(generated)'] = remove_empties(new_config)
+        else:
+            new_config = self.get_snmp_facts()
+            if result['changed']:
+                result['after'] = new_config
+        if self._module._diff:
+            result['diff'] = get_formatted_config_diff(old_config, new_config, self._module._verbosity)
+
+        return result
+
+    def set_config(self, existing_snmp_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = remove_none(self._module.params['config'])
+        have = existing_snmp_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def pop_encrypted_attributes(self, config):
+        """ Remove the encrypted attributes from the config
+        :param config: the config to remove the encrypted attributes from
+        :rtype: None
+        :returns: None
+        """
+        if config:
+            users = config.get('user', [])
+            if users:
+                for user in users:
+                    if user.get('encrypted'):
+                        user.pop('encrypted')
+        return config
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+
+        if state == 'overridden' or state == 'replaced':
+            commands, requests = self._state_replaced_or_overridden(want, have, state)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        return commands, requests
+
+    def _state_replaced_or_overridden(self, want, have, state):
+        """
+        The command generator when state is overridden
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if not want:
+            return commands, requests
+
+        diff_want = get_diff(want, have)
+        if not diff_want:
+            return commands, requests
+
+        del_commands = get_diff(have, want)
+        merged_commands = None
+        merged_request = None
+
+        if del_commands:
+            del_requests = self.get_delete_snmp_request(have, have, True)
```

Comment:
The calculated "del_commands " must be used in this function call.

```suggestion
            del_requests = self.get_delete_snmp_request(del_commands, have, True)
```